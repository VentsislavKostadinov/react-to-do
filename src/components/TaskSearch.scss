.search-task {
    border-radius: 5px !important;
}