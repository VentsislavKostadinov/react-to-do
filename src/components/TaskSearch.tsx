import React, { useState } from 'react'
import { CommonInput } from '../common/CommonInput'
import { TaskSearchProps } from '../model/TaskSearchProps.types'
import './TaskSearch.scss'

export const TaskSearch = ({ filterTasks }: TaskSearchProps) => {
    const [searchTask, setSearchTask] = useState('')

    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const searchText = e.target.value
        setSearchTask(searchText)
        filterTasks(searchText)
    }

    return (
        <CommonInput
            type="search"
            placeholder="Search task"
            value={searchTask}
            handleChange={handleChange}
            className="search-task"
            dataTestid="search-task"
        />
    )
}
