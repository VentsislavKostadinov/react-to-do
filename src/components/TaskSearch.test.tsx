import React from 'react'
import { TaskSearch } from './TaskSearch'
import { TaskSearchProps } from '../model/TaskSearchProps.types'
import { render, screen } from '@testing-library/react'

describe('TaskSearch component', () => {
    const taskSearchProps: TaskSearchProps = {
        filterTasks: jest.fn,
    }

    it('renders correctly', () => {
        const { asFragment } = render(<TaskSearch {...taskSearchProps} />)
        expect(asFragment).toMatchSnapshot()
    })

    it('displayed correctly', () => {
        render(<TaskSearch {...taskSearchProps} />)

        const searchTask = screen.getByTestId('search-task')
        expect(searchTask).toBeInTheDocument
        expect(searchTask).toHaveAttribute('type', 'search')
    })
})
