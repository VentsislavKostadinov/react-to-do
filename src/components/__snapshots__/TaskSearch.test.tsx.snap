// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskSearch component renders correctly 1`] = `[Function]`;
